neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/Neo4jDialect.java

```diff
@@ -634,8 +639,9 @@ public void forEachTuple(ModelConsumer consumer, EntityKeyMetadata... entityKeyM
 		String nativeQuery = buildNativeQuery( backendQuery, queryParameters );
 		Result result = dataBase.execute( nativeQuery, parameters );
 
-		if ( backendQuery.getSingleEntityKeyMetadataOrNull() != null ) {
-			return new NodesTupleIterator( result, backendQuery.getSingleEntityKeyMetadataOrNull() );
+		if ( backendQuery.isSingleEntity() ) {
+			OgmEntityPersister persister = (OgmEntityPersister) sessionFactory.getEntityPersister( backendQuery.getSingleIndexedTypeNameOrNull() );
+			return new NodesTupleIterator( result, backendQuery.getSingleEntityKeyMetadataOrNull(), persister.getTupleContext() );
```

Comment:
Wouldn't make sense to add a parameter to `executeBackendQuery` as well? We will need it anyway in the future to get the id of the running transaction. Something like `QueryContext` (equivalent to `TupleContext`) the can return the additional information that we need.